_crc32c.c

```diff
@@ -49,29 +52,37 @@ PyObject* crc32c_crc32c(PyObject *self, PyObject *args) {
 #else
 	"s*"
 #endif
-	"|I:crc32";
+	"|Ii:crc32";
 
-	if (!PyArg_ParseTuple(args, format, &pbin, &crc) )
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, format, kwlist, &pbin, &crc, &release_gil) )
 		return NULL;
 
 	bin_data = pbin.buf;
-	crc ^= 0xffffffff;
-	result = crc_fn(crc, bin_data, pbin.len);
-	result ^= 0xffffffff;
+	if ((release_gil < 0 && pbin.len >= 32768) || release_gil >= 1) {
```

Comment:
Added a macro called MIN_BUFSIZE_FOR_AUTOMATIC_RELEASE. Renamed `release_gil` to `gil_release_mode`.

Regarding the whole `int | bool` situation: I feel some of the interfaces of this package are a bit un-pythonic right now (i.e. setting the "sw mode" via an environment variable instead of a `kwarg`) and a `gil_release_mode` that maps magic integer values against different behaviour (which you would have to consult documentation for to know what it actually does) feels way more C-esque than pythonic to me.

Reply:
Added a macro called MIN_BUFSIZE_FOR_AUTOMATIC_RELEASE. Renamed release_gil to gil_release_mode.

Regarding the whole `int | bool` situation: I feel some of the interfaces of this package are a bit un-pythonic right now (i.e. setting the "sw mode" via an environment variable instead of a kwarg) and a `gil_release_mode` that maps magic integer values against different behaviour (which you would have to consult documentation for to know what it actually does) feels way more C-esque than pythonic to me. I don't have an opinion on this right now since if you actively want to change this value, you're probably doing some in-depth stuff anyways.

A `release_gil: bool | None = None` still feels way more self-descriptive to me since it describes pretty accurate what's it's doing just from the parameter name and its given typehints already. I did not implement it this way though since it requires some additional boilerplate on the C side, i.e. explicitly checking a `PyObject*` against `Py_False`, `Py_True`, `Py_None` (since `None` is falsy, so one can't simply check for booleaness).

Reply:
Thanks for the further changes!

You are right that using an `int` is an overkill for a flag that chooses between three states. Ideally that would be an enum, but then again that's an even higher overkill given that this is all a C extension -- much easier to deal with a simple `int` that declare a python enum, handle it in C, etc...

Let's continue any further discussion about the type hinting in the other PR. I still think putting the `int | bool` is unnecessary complex (and incidentally the AI bot thing that commented above -- I'm not sure why it's coming up, it probably was enabled in our organisation -- also thought so).

The environment variable is partly fine I think. I do agree that it could be complemented with a kwarg in the `crc32c` function (very easy now that we support them! feel free to add a PR for that :wink:), but it's also a very easy toggle that can be used from any application without code changes.